src/tree/namespace.rs

```diff
@@ -16,6 +17,21 @@ pub struct Namespace {
   pub(crate) ns_ptr: xmlNsPtr,
 }
 
+impl PartialEq for Namespace {
+  fn eq(&self, other: &Self) -> bool {
+    self.get_prefix() == other.get_prefix() && self.get_href() == other.get_href()
+  }
+}
+
+impl Eq for Namespace {}
+
+impl Hash for Namespace {
+  fn hash<H: Hasher>(&self, state: &mut H) {
+    self.get_prefix().hash(state);
+    self.get_href().hash(state);
+  }
+}
```

Comment:
Similarly to above, using the pointer directly could be another reasonable approach to hashing.